src/shared/mount-util.c

```diff
@@ -1441,6 +1441,23 @@ int make_userns(uid_t uid_shift,
         return TAKE_FD(userns_fd);
 }
 
+int open_tree_attr_fallback(int dir_fd, const char *path, unsigned int flags, struct mount_attr *attr, size_t size) {
```

Comment:
You might as well just supply sizeof(struct mount_attr) internally? No need for size param?

assert(attr)